hmy/api_backend.go

```diff
@@ -139,7 +139,16 @@ func (b *APIBackend) ProtocolVersion() int {
 // GetLogs ...
 func (b *APIBackend) GetLogs(ctx context.Context, blockHash common.Hash) ([][]*types.Log, error) {
 	// TODO(ricl): implement
-	return nil, nil
+	// return nil, nil
+	receipts := b.hmy.blockchain.GetReceiptsByHash(blockHash)
+	if receipts == nil {
+		return nil, nil
+	}
+	logs := make([][]*types.Log, len(receipts))
+	for i, receipt := range receipts {
+		logs[i] = receipt.Logs
+	}
+	return logs, nil
```

Comment:
why copy?

Reply:
Pointers?

Reply:
Is it a copy? Wouldn't the receipt.Logs array be passed by reference as []*types.Log at logs[i]?

Implementation borrowed from ethereum: https://github.com/ethereum/go-ethereum/blob/d90d1db609c8d77baa422d49bd371207c06b4711/eth/api_backend.go#L177

Doesn't mean it's right.